library/core/src/num/nonzero.rs

```diff
@@ -490,19 +489,20 @@ macro_rules! nonzero_integer {
         ///
         /// [null pointer optimization]: crate::option#representation
         #[$stability]
+        #[deprecated(since = "TBD", note = "replaced by the generic `NonZero` type")]
```

Comment:
I don't think a deprecation of the NonZero* type aliases has been approved by the standard library team. It would be better to land all the other changes that can be landed without this deprecation (such as docs) as soon as possible, and move this deprecation into a separate PR for a slower team-wide consensus.

Reply:
I also feel like it may be good to wait a few releases before deprecating, so that crates with a MSRV policy of stable-N for small N will fix the deprecation instead of allowing the lint (and then forgetting to fix the deprecation later).

Reply:
> wait a few releases before deprecating

Isn't that what `deprecated_in_future` is meant for?

In any case I will move it into a separate PR.